### DISCUSSION TITLE
เราจะยืนยันตัวตนใน Discord ก้าว Geek กันยังไงดี

### DISCUSSION THREAD

**Discussion by heypoom** [kaogeek/kaogeek-discord-bot#3] [Ideas]
ลองมาเสนอวิธียืนยันตัวตน (เช่น P2P Verification, Digital ID) กัน

**Comment by Tapao-NonSen** [comment]
ยืนยันหลังจากเข้าเซิฟเวอร์เพื่อรับยศสำหรับการพิพม์เเชทหรือดูในห้องต่างๆ ครับโดยการใช้ ReCAPTCHA 
อย่าง https://altdentifier.com/

**Comment by tnwisme** [comment]
อยากให้ใช้เหมือนกับ ReCaptcha ครับ
เวลาโดน Raid บอทมันอาจจะกด Reaction Role เองได้ถ้าใช้ ReCaptcha น่าจะดีกว่าครับ

**Comment by opecgame** [comment]
การ Verify เนี่ยครับผมอยากเสนอให้เป็นการที่กรอกข้อมูลส่วนตัวบางอย่าง ลงไปใน Form + reCaptcha หรือ CF Turnsite โดยและจะได้ยศ Verify ใน Discord ครับ

**Comment by GmBeHappy** [comment]
ผู้ใช้งานมาผ่านเว็บ 9geek -> เข้า discord ใหม่ มาใน role ที่อาจมีการจำกัดสิทธ์เช่น อ่านได้บาง chanel พิมพ์ไม่ได้ -> (ถ้าต้องการ up role) -> เลือก role ที่ต้องการจะ verify (dev , contributor, etc) -> มี bot จาก discord ส่ง link สำหรับ verify ที่มีเวลาจำกัดการใช้ link (redirect) -> ไปเว็บ 9geek เพื่อ verify ผ่าน OAuth | etc (อาจมีการใช้คนหหรืออื่นๆตรวจสอบ) (สามารถเพิ่ม capcha หรืออื่นๆได้) -> แล้วถ้า verify ผ่านแล้วให้ bot ปรับ role ให้
  ที่ต้องมี  
  - หลังบ้านของเว็บ 9geek 
  - bot discord

ถ้าเปิดเป็น open source แล้วให้มีคนเข้ามาช่วย contribute น่าจะเกิดขึ้นได้ครับ

**Reply by mildronize** [reply]
@GmBeHappy น่าสนใจมากคับ

**Reply by borwornpob** [reply]
ในส่วนของเว็บ 9geek ควรจะตั้ง Repo ใหม่แล้วทำให้เป็น Open Source ขึ้นมาเลยมั้ยครับ ในส่วนตัวมองว่าอาจจะเป็น Hub ของ services ต่างๆที่ในอนาคตจะเกิดขึ้น + พวก API endpoints จะได้มาอยู่ที่เดียวกัน

**Reply by damp11113** [reply]
น่า ok อยู่นะ
